src/sdk/namespace_blob.js

```diff
@@ -254,13 +254,19 @@ class NamespaceBlob {
         );
         let obj;
         const blob_client = this._get_blob_client(params.key);
+        const tagging = params.tagging && Object.fromEntries(params.tagging.map(tag => ([tag.key, tag.value])));
         if (params.copy_source) {
             if (params.copy_source.ranges) {
                 throw new Error('NamespaceBlob.upload_object: copy source range not supported');
             }
-            const sasToken = await this._get_sas_token(params.copy_source.bucket, params.copy_source.key, 'r'); // read permission
+            const tag_permision = params.tagging_copy || params.tagging ? 't' : null;
+            const sasToken = await this._get_sas_token(params.copy_source.bucket, params.copy_source.key, 'r' + tag_permision); // read permission
```

Comment:
according to node: 
'r' + null = 'rnull'
that's not what you would like to get, you should replace null with ''